AppInsightsWithWebJob/Program.cs

```diff
@@ -35,16 +38,16 @@ static async Task Main(string[] args)
             });
 
             var host = builder.Build();
-            var cancellationToken = new WebJobsShutdownWatcher().Token;
 
             var telemetryClient = host.Services.GetService<TelemetryClient>();
-            AppDomain.CurrentDomain.UnhandledException += (sender, eventArgs) => telemetryClient.TrackException((Exception)eventArgs.ExceptionObject);
-            TaskScheduler.UnobservedTaskException += (sender, eventArgs) => telemetryClient.TrackException((Exception)eventArgs.Exception);
 
-            using (cancellationToken.Register(() => host.Services.GetService<IJobHost>().StopAsync().GetAwaiter().GetResult()))
-            using (host)
+            try
+            {
+                await host.RunAsync();
+            }
+            catch (Exception exception)
             {
-                await host.RunAsync(cancellationToken);
+                telemetryClient.TrackException(exception);
```

Comment:
This is good for errors on startup but you also need to handle errors in the background service itself.

Reply:
Makes sense. Doing that now.